src/test/java/core/basesyntax/service/impl/TransactionProcessingServiceImplTest.java

```diff
@@ -0,0 +1,81 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionProcessingServiceImplTest {
+    private static Map<FruitTransaction.Operation,
+            OperationCalculator> countStrategyMap;
+    private static OperationCalculatorStrategy operationStrategy;
+    private static TransactionProcessingService transactionProcessingService;
+    private static Storage storage;
+
+    @BeforeClass
+    public static void setUp() {
+        countStrategyMap = Map.of(
+                FruitTransaction.Operation.BALANCE, new BalanceCountStrategyImpl(),
+                FruitTransaction.Operation.SUPPLY, new SupplyCountStrategyImpl(),
+                FruitTransaction.Operation.PURCHASE, new PurchaseCountStrategyImpl(),
+                FruitTransaction.Operation.RETURN, new ReturnCountStrategyImpl());
+        operationStrategy
+                = new OperationCalculatorStrategy(countStrategyMap);
+        transactionProcessingService
+                = new TransactionProcessingServiceImpl(operationStrategy);
+        storage = new Storage();
+    }
+
+    @AfterClass
+    public static void cleanStorage() {
+        storage.getFruitMap().clear();
+    }
+
+    @Test
+    public void update_validInput_ok() {
+        List<FruitTransaction> input = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 110),
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 230),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 10),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "apple", 90),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "banana", 10),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 90),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 10),
+                new FruitTransaction(FruitTransaction.Operation.SUPPLY, "apple", 90)
+        );
+        transactionProcessingService.update(input);
+        Assert.assertTrue("invalid result of updating fruitMap: "
+                        + "expected value for: " + "banana"
+                        + " = " + 120,
+                storage.getFruitMap().containsKey("banana")
+                        && storage.getFruitMap().get("banana") == 120);
+        Assert.assertTrue("invalid result of updating fruitMap: "
+                        + "expected value for: " + "apple"
+                        + " = " + 320,
+                storage.getFruitMap().containsKey("apple")
+                        && storage.getFruitMap().get("apple") == 320);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void update_nullInput_notOk() {
+        transactionProcessingService.update(null);
+    }
+
+    @Test(expected = FruitTransactionException.class)
+    public void update_emptyInput_notOk() {
+        transactionProcessingService.update(Collections.emptyList());
```

Comment:
IMHO, it looks strange that we throw an exception if try to update storage by an empty list. I expect that nothing will happen. 
For example, your app can call this method once in an hour, and if during this hour there were no added any transaction, your app will stop.